CodeGeneration/Sources/SyntaxSupport/AvailabilityNodes.swift

```diff
@@ -118,42 +118,54 @@ public let AVAILABILITY_NODES: [Node] = [
     ]
   ),
 
-  // version-tuple -> integer-literal
-  //                | float-literal
-  //                | float-literal '.' integer-literal
+  // version-tuple-element -> '.' float-literal |
+  //                          '.' interger-literal
+
   Node(
-    name: "VersionTuple",
-    nameForDiagnostics: "version tuple",
-    description: "A version number of the form major.minor.patch in which the minor and patch part may be omitted.",
+    name: "VersionTupleElement",
+    nameForDiagnostics: "version tuple element",
+    description: "An element to represent a dot and number pair",
     kind: "Syntax",
     children: [
       Child(
-        name: "Major",
-        kind: .token(choices: [.token(tokenKind: "IntegerLiteralToken")]),
-        description: "The major version."
-      ),
-      Child(
-        name: "MinorPeriod",
+        name: "period",
```

Comment:
These names should be capitalized. Otherwise we get a lowercase name in the unexpected names. 

And since you just hit this. Would you like to also add an assertion that the first letter of a child’s name is uppercase?